### DISCUSSION TITLE
Sharing Allow/Block Lists between Nodes

### DISCUSSION THREAD

**Discussion by bem4444** [libp2p/rust-libp2p#4881] [Q&A] [answered]
I'm trying to use the allow block list as part of my Network Behaviour and was curious about how I could potentially share this information between nodes. As far as I can tell the way that allow_block_list is implemented there is no way to tell if a Peer is present in an AllowedPeers list. How would one node query another node to see if a Peer is in its AllowedPeers list or not?

**Comment by dariusc93** [comment]
The best way to do this is to exchange messages either through floodsub, gossipsub or your own protocol. In your case, you could use `libp2p-request-response` to request or query the node and provide a response from that node

**Reply by bem4444** [reply]
That makes sense. I guess I meant more how would a node know whether a Peer is in its own AllowedPeers list or not. If I understand correctly that struct is private and the allow_block_list Behaviour doesn't have access to it. Is that a right understanding or am I missing something?

**Reply by thomaseizinger** [reply, marked as answer]
`libp2p-allow-block-list` is just a module that enforces your allow/block list policy. It is not the source of truth for your allow-block list as you need to first add all the peers to it.

Thus, whatever components calls the API of `libp2p-allow-block-list` is in charge of managing that state and it can also distribute it to other behaviours or answer incoming requests about it.

Long-story short: It is the application's job to manage this list and update `libp2p-allow-block-list` with any changes to it.
